examples/offline_inference_neuron.py

```diff
@@ -32,7 +32,9 @@
     # The device argument can be either unspecified for automated detection,
     # or explicitly assigned.
     device="neuron",
-    tensor_parallel_size=2)
+    tensor_parallel_size=2,
+    # Single node inference
+    world_size=1)
```

Comment:
I'm confuse with world_size, while TP size=2. Isn't world_size predictable from number of nodes times TP size?

Reply:
> Compared to single node tensor parallel, multi-node tensor parallel shards the model weights in the same way but having mores cores across nodes. In the meantime, it requires each node’s model.forward() receives the exact same input, otherwise there would be unexpected behaviors (runtime failure, wrong output).

For Neuron multi node inference, we need to broadcast the API input params across all the nodes. So the world size for Neuron within the scope of Vllm is the number of nodes on which we perform inference instead of TP*PP as model parallelism is handled within the context of neuron (outside the scope of vllm)

Reply:
we can still reuse existing vLLM config argument to calculate world size, right?

the equation looks like:
```
world_size = tensor_parallel_size * pipeline_parallel_size
```

Reply:
If we use the existing flow of using world_size = TP*PP then it would come out to be 64 for a cluster of 2 trn1/trn1n instances. This causes the init_process_group to hang as we only initialize it with 2 ranks (0 and 1 from each node of the cluster) instead of all the ranks.

This was currently working for single node inference with TP=32 because we had hard coded the world size to 1 while initializing the process groups [here](https://github.com/vllm-project/vllm/blob/main/vllm/worker/neuron_worker.py#L118) instead of using world_size = TP*PP which would be 32.

Reply:
we can send fake input to `init_process_group` so that we won't get hang there.